x/evm/types/msg.go

```diff
@@ -77,16 +77,7 @@ func newMsgEthereumTx(
 	}
 
 	switch {
-	case tx.Accesses == nil:
-		txData = &LegacyTx{
-			To:       toAddr,
-			Amount:   amt,
-			GasPrice: gp,
-			Nonce:    tx.Nonce,
-			GasLimit: tx.GasLimit,
-			Data:     tx.Input,
-		}
-	case tx.Accesses != nil && tx.GasFeeCap != nil && tx.GasTipCap != nil:
+	case tx.GasFeeCap != nil:
```

Comment:
what about the tip cap and access list not being nil?

Reply:
AccessList are not necessary for an DynamicFeeTx. This had always been wrong. I changed it to be in sync with way Geth classifies the txs https://github.com/ethereum/go-ethereum/blob/e5eb32acee19cc9fca6a03b10283b7484246b15a/internal/ethapi/transaction_args.go#L272

Reply:
can you check in the types instead of the RPC? https://github.com/ethereum/go-ethereum/blob/0004c6b229b787281760b14fb9460ffd9c2496f1/core/types/transaction.go